app/models/conversation.rb

```diff
@@ -163,6 +137,53 @@ def self.unreachable_phone_error(errormsg)
     errormsg.is_a?(String) && errormsg =~ /30003|30005|30006/
   end
 
+
+  def api_json(include_messages = false)
+    fields = [:id, :user_id, :pickup_at, :status, :lifecycle, :from_phone,
+              :from_latitude, :from_longitude, :to_latitude, :to_longitude, :additional_passengers, :blacklisted_phone_number]
+    j = self.as_json(only: fields, methods: [:message_count])
+    if include_messages
+      j['messages'] = self.messages.map(&:api_json)
+    end
+    %w(from_address from_city to_address to_city special_requests ).each do |field|
+      j[field] = CGI::escape_html(send(field) || '')
+    end
+    last_msg = self.messages.order('created_at ASC').last
+    if last_msg
+      j['last_message_time'] = last_msg.created_at.to_i
+      j['last_message_sent_by'] = last_msg.sent_by
+      j['last_message_body'] = CGI::escape_html(last_msg.body || '')
+    end
+    j['from_phone'] = self.from_phone.phony_formatted(normalize: :US, spaces: '-')
+    j['status_updated_at'] = self.status_updated_at.to_i
+    j['name'] = CGI::escape_html(username || '')
+    j['ride'] = ride.api_json if ride
+    j['blacklisted_phone_number'] = self.blacklisted_phone.phone if self.blacklisted_phone.present?
+    j
+  end
+
+  # have language and name and confirmed origin and destination and confirmed time and passengers
+  def has_fields_for_ride
+    if self.user.blank? ||
+      # self.user.name.blank? ||
+      self.from_address.blank? ||
+      self.from_city.blank? ||
+      self.from_latitude.blank? ||
+      self.from_longitude.blank? ||
+      self.pickup_at.blank?
+      false
+    else
+      true
+    end
+  end
```

Comment:
Could also be written
```rb
def has_fields_for_ride
  user.present? &&
  from_address.present? &&
  from_city.present? &&
  from_latitude.present? &&
  from_longitude.present? &&
  pickup_at.present?
end
```
Also: predicate methods should probably end in `?`

Reply:
thanks much better, done